app/cdap/components/PipelineTriggers/PipelineListTab/index.tsx

```diff
@@ -0,0 +1,399 @@
+import { useFeatureFlagDefaultTrue } from 'services/react/customHooks/useFeatureFlag';
+import IconButton from '@material-ui/core/IconButton';
+import Button from '@material-ui/core/Button';
+import DeleteIcon from '@material-ui/icons/Delete';
+import CloseIcon from '@material-ui/icons/Close';
+import {
+  IPipelineInfo,
+  IProgramStatusTrigger,
+  ISchedule,
+} from 'components/PipelineTriggers/store/ScheduleTypes';
+import {
+  PipelineListHeader,
+  PipelineName,
+  PipelineTriggerHeader,
+} from 'components/PipelineTriggers/shared.styles';
+import { TextField } from '@material-ui/core';
+
+const TRIGGER_PREFIX = 'features.PipelineTriggers';
+const PREFIX = `${TRIGGER_PREFIX}.SetTriggers`;
+
+const CloseTabIconButton = styled(IconButton)`
+  float: right;
+`;
+
+const EnableGroupTriggerButton = styled(Button)`
+  background-color: #5a84e4;
+  text-transform: none;
+  margin: 10px 0;
+`;
+
+const NamespaceSelectorDropdown = styled.div`
+  display: inline-block;
+  width: 150px;
+  margin-left: 10px;
+  border-bottom: 2px solid #333333;
+  margin-bottom: 10px;
+
+  select.form-control:not([size]):not([multiple]) {
+    border: 0;
+    height: initial;
+    box-shadow: none;
+    cursor: pointer;
+  }
+`;
+
+const PipelineCount = styled.div`
+  margin-bottom: 10px;
+`;
+
+const PipelineListTabDiv = styled.div`
+  padding: 15px;
+`;
+
+const SelectedGroupPipelinesContainer = styled.div`
+  margin-bottom: 20px;
+  font-weight: bold;
+`;
+
+const StyledDeleteIconButton = styled(IconButton)`
+  && {
+    padding: 7px;
+  }
+`;
+
+const TypeSelectorDropdown = styled(NamespaceSelectorDropdown)`
+  width: 255px;
+`;
+
+const TriggerNameTextField = styled(TextField)`
+  width: 255px;
+  && {
+    margin: 5px 0 10px 5px;
+    vertical-align: inherit;
+    .MuiInput-underline:after {
+      border-bottom: 2px solid black;
+    }
+    .MuiInputBase-input {
+      text-align: center;
+    }
+    .MuiFormHelperText-root.Mui-error {
+      text-align: center;
+    }
+  }
+`;
+
+interface IPipelineListTabViewProps {
+  existingTriggers: ISchedule[];
+  pipelineList: IPipelineInfo[];
+  triggersGroupToAdd: IProgramStatusTrigger[];
+  triggersGroupRunArgsToAdd: any;
+  selectedNamespace: string;
+  selectedTriggersType: string;
+  pipelineName: string;
+  expandedPipeline: string;
+  toggleExpandPipeline: (pipeline: string) => void;
+  workflowName: string;
+  configureError: string;
+  onPayloadToggle: (isOpen: boolean) => void;
+  payloadModalIsOpen: boolean;
+  setTab: (tab: number) => void;
+}
+
+const PipelineListTabView = ({
+  existingTriggers,
+  pipelineList,
+  triggersGroupToAdd,
+  triggersGroupRunArgsToAdd,
+  pipelineName,
+  selectedNamespace,
+  selectedTriggersType,
+  expandedPipeline,
+  toggleExpandPipeline,
+  workflowName,
+  configureError,
+  onPayloadToggle,
+  payloadModalIsOpen,
+  setTab,
+}: IPipelineListTabViewProps) => {
+  const [namespaceList, setNamespaceList] = useState([]);
```

Comment:
nit: I general I think when there are more than 3 useState calls I replace it with a useReducer